pkg/expr/expr_types.go

```diff
@@ -23,6 +23,26 @@ type (
 		Type
 		ResolveTypes(func(string) Type) error
 	}
+
+	iterator interface {
+		Each(func(string, interface{}) error) error
+	}
+
+	setter interface {
```

Comment:
could an existing `FieldAssigner` interface be used instead of this, new one?

Reply:
Yes